runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java

```diff
@@ -244,6 +266,22 @@ public void close() throws Exception
         }
     }
 
+    private void reconfigure(URL configURL, String configText)
+    {
+        try
+        {
+            NamespaceConfig namespace = namespaces.get(configURL);
+            unregisterTask.setRootNamespace(namespace);
+            commonPool().submit(unregisterTask).get();
+            registerTask.setConfigText(configText);
+            namespaces.put(configURL, commonPool().submit(registerTask).get());
```

Comment:
I actually like having seperate tasks. If we'd put these parts in the engine, it'd have lots of fields that are only necessary for configuration. I think it's also beneficial that now the engine is (almost) only responsible for lifecycle management of Zilla (start, close). Other than those, it creates the tasks, and let them handle the configuration part. But maybe merging register + unregister is worth considering, since they're always called together now. 

Also if you're not convinced, I'm open to change these to a simple function :)

Reply:
It might be helpful to combine register + unregister and it's not really as task any more, no need to submit on another thread, perhaps just a helper class to manage the relevant state, single instance, simple method call to trigger behavior when needed.

Reply:
I like that idea. I created a ConfigManager class for this.